### PR TITLE
Adding proper fsync errors signaling for pipes.

### PR DESCRIPTION
**Pull request by abujalski**
This change sets properly errno flag for fsync function called on fd representing pipe.

This change fixes http://posixtest.sourceforge.net/ fsync/7-1 test.

### CODE REVIEW

**Comment by kripken** [comment]
This seems to have broken CI. Oddly no tests were run on it here, I'm not sure why? (I don't see `[ci skip]` or anything.)

After it landed on incoming, tests broke there, and the error message mentions `fsync`.

If we can't fix this quickly, we can revert it.

File: src/library_syscall.js [hunk 1]
```diff
@@ -710,6 +710,9 @@ var SyscallsLibrary = {
       });
     });
 #else
+    if (stream.stream_ops.fsync) {
+        return -stream.stream_ops.fsync(stream);
```
Comment (kripken) [inline src/library_syscall.js]: indentation looks wrong here - should be 2 spaces, not 4.

We need to fix CI for this PR anyhow, can fix the indentation

Comment (sbc100) [reply src/library_syscall.js]: Very strange.    I'll guess I'll revert.  @abujalski can you re-upload a new PR?

Comment (sbc100) [reply src/library_syscall.js]: Reverted in 771fd6b27

**Pull request merged**